### ISSUE TITLE
Nested include breaks CacheStore#setCollection

### ISSUE THREAD

**Issue by xaun** [reyesoft/ngx-jsonapi#128]
Is the nested include supported? 

```
.all({ include: 'comments.author' })
```

On version 1.1.1 when trying to query with the above it breaks with stack trace: 

```
marketplace.component.ts:27 TypeError: Cannot read property 'data' of undefined
    at eval (ngx-jsonapi.es5.js:1956)
    at Array.forEach (<anonymous>)
    at eval (ngx-jsonapi.es5.js:1955)
    at eval (ngx-jsonapi.es5.js:351)
    at Array.forEach (<anonymous>)
    at Function.Base.forEach (ngx-jsonapi.es5.js:350)
    at CacheStore.setCollection (ngx-jsonapi.es5.js:1952)
    at eval (ngx-jsonapi.es5.js:2355)
    at ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:4760)
```

The `resource_type_alias` is `"comments.author"`, so `resource.relationships[resource_type_alias].data` can never work.

**Comment by jonasgrundtvig** [comment]
The exact same problem occurs in version 2.0.0.-rc.10.

**Comment by maxi7587** [comment]
@pablorsk Do you want me to take a look?

**Comment by pablorsk** [comment]
Yeah, you're right. (https://jsonapi.org/format/#fetching-includes)

http://jsonapiplayground.reyesoft.com/ dont support yet compound includes, I can't test.

**Comment by stefandb** [comment]
I made a PR for this issue to fix this. https://github.com/reyesoft/ngx-jsonapi/pull/140

**Comment by pablorsk** [comment]
PR  #140 and PR #169 try to fix this problem. @maxi7587 take this issue on FE-4 (internal issue code of @reyesoft/reyesoft-team)

**Comment by maxi7587** [comment]
I'm on it!

**Comment by xaun** [comment]
@maxi7587 @pablorsk Has this been done? I can see #175 was merged, but I'm trying to test now on v2.0.2 and while it's not throwing errors, the nested include relationship resource has an empty id and empty attributes (in the network request, the relationships are included). 

Something else I have noticed is that the while the DocumentCollection response of my `all` request `is_loaded` is `true`, `is_loaded` value for all nested relationships is `false`.

In the screenshot below you can see the nested relationships which lead to the empty "fields" resource.
<img width="791" alt="Screen Shot 2019-05-14 at 13 11 00" src="https://user-images.githubusercontent.com/7486451/57668497-e1d4fc00-764a-11e9-8bb0-4e583f6d5c41.png">

My services are registered and injected (no register warnings).

My models are defined correctly I believe 
```typescript
export class Product extends Resource {
  public relationships = {
    values: new DocumentCollection<ProductValue>()
  }
}

export class ProductValue extends Resource {
  public relationships = {
    field: new DocumentResource<ProductValueField>()
  }
}

export class ProductValueField extends Resource {
  attributes: {
    created_at: string
    description: string
    field_type: string
    max_length: number | null
    min_length: number | null
    multiple: boolean
    name: string
    nid: string
    position: number
    required: boolean
    section: string
    updated_at: string
    visible: boolean
  }
}

@Injectable({
  providedIn: 'root'
})
@Autoregister()
export class ProductService extends Service<Product> {
  public resource = Product
  public type = 'products'

  constructor(
    private productValueService: ProductValueService,
    private productValueFieldService: ProductValueFieldService) {
    super()

    this.all({ include: ['values.field'] }).pipe(filter(res => !res.is_loaded)).subscribe(res => {
      console.log(res) // this is the screenshotted output
    })
  }
}

@Injectable({
  providedIn: 'root'
})
@Autoregister()
export class ProductValueService extends Service<ProductValue> {
  public resource = ProductValue
  public type = 'values'

  constructor() {
    super()
  }
}

@Injectable({
  providedIn: 'root'
})
@Autoregister()
export class ProductValueFieldService extends Service<ProductValueField> {
  public resource = ProductValueField
  public type = 'fields'

  constructor() {
    super()
  }
}
```

Am I missing something? Thanks

**Comment by karwank** [comment]
@xaun please look what I did in #180, maybe it will help you.

**Comment by pablorsk** [comment]
Solved on v2.1.9

**Comment by noahott** [comment]
I'm having issues with this still in v2.1.10. Attributes in nest relationships are not always loaded:

![Screen Shot 2019-12-30 at 2 49 21 PM](https://user-images.githubusercontent.com/1277799/71598135-b2103100-2b13-11ea-8799-12c5f9c51ca5.png)
![Screen Shot 2019-12-30 at 2 51 33 PM](https://user-images.githubusercontent.com/1277799/71598206-eedc2800-2b13-11ea-80fb-234fae4555db.png)
